components/renku_data_services/session/apispec_base.py

```diff
@@ -12,8 +14,10 @@ class Config:
 
         from_attributes = True
 
-    @field_validator("id", mode="before", check_fields=False)
+    @field_validator("*", mode="before", check_fields=False)
     @classmethod
-    def serialize_id(cls, id: str | ULID) -> str:
-        """Custom serializer that can handle ULIDs."""
-        return str(id)
+    def serialize_ulid(cls, value: Any) -> Any:
+        """Handle ULIDs."""
+        if isinstance(value, ULID):
```

Comment:
that's a nice way to handle it :+1: , we might want to add that for the other apispec_base.py's as well

Reply:
I was thinking that we should have a common `apispec_base.py` file because otherwise we have a lot of code fragmentation.